cmake/process_options.cmake

```diff
@@ -141,7 +143,62 @@ if(ENABLE_KOKKOS)
         message(STATUS "Found existing Kokkos Kernels library")
     endif()
 
+    # If the package cannot be found, explicitly search for the built libs and headers
     if (NOT (Kokkos_FOUND AND KokkosKernels_FOUND))
+        find_library(kokkos_core_lib
+            NAME kokkoscore.a libkokkoscore.a kokkoscore.so libkokkoscore.so
+            HINTS   ${CMAKE_SOURCE_DIR}/Kokkos/lib
+                    ${Kokkos_Core_DIR}/lib
+                    ${Kokkos_Core_DIR}/lib64
+                    /usr/lib
+                    /usr/lib64
+                    /usr/local/lib
+                    /usr/local/lib64
+                    ENV LD_LIBRARY_PATH
+        )
+        find_library(kokkos_kernels_lib
+            NAME kokkoskernels.a libkokkoskernels.a kokkoskernels.so libkokkoskernels.so
+            HINTS   ${CMAKE_SOURCE_DIR}/Kokkos/lib
+                    ${Kokkos_Kernels_DIR}/lib
+                    ${Kokkos_Kernels_DIR}/lib64
+                    /usr/lib
+                    /usr/lib64
+                    /usr/local/lib
+                    /usr/local/lib64
+                    ENV LD_LIBRARY_PATH
+        )
+        find_file(  kokkos_core_inc
+            NAMES   Kokkos_Core.hpp
+            HINTS   ${Kokkos_Core_DIR}/include
+                    /usr/include
+                    /usr/local/include
+                    ENV CPATH
+        )
+        find_file(  kokkos_sparse_inc
+            NAMES   KokkosSparse.hpp
+            HINTS   ${Kokkos_Kernels_DIR}/include
+                    /usr/include
+                    /usr/local/include
+                    ENV CPATH
+        )
+    endif()
+
+    # Check if the package has been found again
+    if (kokkos_core_lib_FOUND AND kokkos_kernels_lib_FOUND)
+        add_library( kokkos SHARED IMPORTED GLOBAL)
+        add_library( kokkoskernels SHARED IMPORTED GLOBAL)
+
+        cmake_path(GET kokkos_core_inc ROOT_PATH kokkos_INC_DIR)
+        cmake_path(GET kokkos_sparse_inc ROOT_PATH kokkoskernels_INC_DIR)
+
+        set_target_properties( kokkos PROPERTIES IMPORTED_LOCATION ${kokkos_core_lib})
+        set_target_properties( kokkoskernels PROPERTIES IMPORTED_LOCATION ${kokkos_kernels_lib})
+        set_target_properties( kokkos PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${kokkos_INC_DIR}")
+        set_target_properties( kokkoskernels PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${kokkoskernels_INC_DIR}")
+        target_link_libraries( lightning_external_libs INTERFACE kokkos kokkoskernels)
```

Comment:
In the case of finding Kokkos, this linking step will only work if Kokkos was found on the second try when looking for the files.